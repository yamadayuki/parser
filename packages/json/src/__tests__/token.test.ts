import { Lexer, ILexerConfig } from "chevrotain";
import {
  True,
  False,
  Null,
  LCurly,
  RCurly,
  LSquare,
  RSquare,
  Comma,
  Colon,
  StringLiteral,
  NumberLiteral,
  WhiteSpace,
} from "../token";

const lexerOption: ILexerConfig = { positionTracking: "onlyOffset" };

describe("token", () => {
  test("true", () => {
    const trueLexer = new Lexer([True], lexerOption);
    expect(trueLexer.tokenize("true").tokens).toMatchSnapshot();
  });

  test("false", () => {
    const falseLexer = new Lexer([False], lexerOption);
    expect(falseLexer.tokenize("false").tokens).toMatchSnapshot();
  });

  test("null", () => {
    const nullLexer = new Lexer([Null], lexerOption);
    expect(nullLexer.tokenize("null").tokens).toMatchSnapshot();
  });

  test("LCurly", () => {
    const lCurlyLexer = new Lexer([LCurly], lexerOption);
    expect(lCurlyLexer.tokenize("{").tokens).toMatchSnapshot();
  });

  test("RCurly", () => {
    const rCurlyLexer = new Lexer([RCurly], lexerOption);
    expect(rCurlyLexer.tokenize("}").tokens).toMatchSnapshot();
  });

  test("LSquare", () => {
    const lSquareLexer = new Lexer([LSquare], lexerOption);
    expect(lSquareLexer.tokenize("[").tokens).toMatchSnapshot();
  });

  test("RSquare", () => {
    const rSquareLexer = new Lexer([RSquare], lexerOption);
    expect(rSquareLexer.tokenize("]").tokens).toMatchSnapshot();
  });

  test("Comma", () => {
    const commaLexer = new Lexer([Comma], lexerOption);
    expect(commaLexer.tokenize(",").tokens).toMatchSnapshot();
  });

  test("Colon", () => {
    const colonLexer = new Lexer([Colon], lexerOption);
    expect(colonLexer.tokenize(":").tokens).toMatchSnapshot();
  });

  test("StringLiteral", () => {
    const stringLiteralLexer = new Lexer([StringLiteral], lexerOption);
    expect(stringLiteralLexer.tokenize('"foo"').tokens).toMatchSnapshot();
  });

  test("NumberLiteral", () => {
    const numberLiteralLexer = new Lexer([NumberLiteral], lexerOption);
    expect(numberLiteralLexer.tokenize("123").tokens).toMatchSnapshot();
  });

  test("WhiteSpace", () => {
    const whiteSpaceLexer = new Lexer([WhiteSpace], lexerOption);
    expect(whiteSpaceLexer.tokenize("\t").tokens).toMatchSnapshot();
  });
});
