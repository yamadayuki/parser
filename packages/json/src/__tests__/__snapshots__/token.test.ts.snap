// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`token Colon 1`] = `
Array [
  Object {
    "image": ":",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": ":",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "Colon",
      "tokenTypeIdx": 10,
    },
    "tokenTypeIdx": 10,
  },
]
`;

exports[`token Comma 1`] = `
Array [
  Object {
    "image": ",",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": ",",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "Comma",
      "tokenTypeIdx": 9,
    },
    "tokenTypeIdx": 9,
  },
]
`;

exports[`token LCurly 1`] = `
Array [
  Object {
    "image": "{",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": "{",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "LCurly",
      "tokenTypeIdx": 5,
    },
    "tokenTypeIdx": 5,
  },
]
`;

exports[`token LSquare 1`] = `
Array [
  Object {
    "image": "[",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": "[",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "LSquare",
      "tokenTypeIdx": 7,
    },
    "tokenTypeIdx": 7,
  },
]
`;

exports[`token NumberLiteral 1`] = `
Array [
  Object {
    "image": "123",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": /-\\?\\(0\\|\\[1-9\\]\\\\d\\*\\)\\(\\\\\\.\\\\d\\+\\)\\?\\(\\[eE\\]\\[\\+-\\]\\?\\\\d\\+\\)\\?/,
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "NumberLiteral",
      "tokenTypeIdx": 12,
    },
    "tokenTypeIdx": 12,
  },
]
`;

exports[`token RCurly 1`] = `
Array [
  Object {
    "image": "}",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": "}",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "RCurly",
      "tokenTypeIdx": 6,
    },
    "tokenTypeIdx": 6,
  },
]
`;

exports[`token RSquare 1`] = `
Array [
  Object {
    "image": "]",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": "]",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "RSquare",
      "tokenTypeIdx": 8,
    },
    "tokenTypeIdx": 8,
  },
]
`;

exports[`token StringLiteral 1`] = `
Array [
  Object {
    "image": "\\"foo\\"",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": /"\\(\\?:\\[\\^\\\\\\\\"\\]\\|\\\\\\\\\\(\\?:\\[bfnrtv"\\\\\\\\\\\\/\\]\\|u\\[0-9a-fA-F\\]\\{4\\}\\)\\)\\*"/,
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "StringLiteral",
      "tokenTypeIdx": 11,
    },
    "tokenTypeIdx": 11,
  },
]
`;

exports[`token WhiteSpace 1`] = `Array []`;

exports[`token false 1`] = `
Array [
  Object {
    "image": "false",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": "false",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "False",
      "tokenTypeIdx": 3,
    },
    "tokenTypeIdx": 3,
  },
]
`;

exports[`token null 1`] = `
Array [
  Object {
    "image": "null",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": "null",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "Null",
      "tokenTypeIdx": 4,
    },
    "tokenTypeIdx": 4,
  },
]
`;

exports[`token true 1`] = `
Array [
  Object {
    "image": "true",
    "startOffset": 0,
    "tokenType": Object {
      "CATEGORIES": Array [],
      "PATTERN": "true",
      "categoryMatches": Array [],
      "categoryMatchesMap": Object {},
      "isParent": false,
      "tokenName": "True",
      "tokenTypeIdx": 2,
    },
    "tokenTypeIdx": 2,
  },
]
`;
